Simplify location of protobuf legal license files

mdedetrich opened the pull request:
The current location of exception cases for legal license files are somewhat confusing, i.e. the root `LICENSE` file is actually a license that should only be used for the sbt protobuf subprojects.

This PR makes very clear/simple about which LICENSE files are used where, i.e.

* `LICENSE`: This file is added as `LICENSE` to `META-INF` for every sbt sub project apart from the protobuf ones (i.e. its the global default)
* `legal/LICENSE.protobuf`: This overrides the root `LICENSE` file and hence is added as `LICENSE` to the `META-INF` of protobuf projects
* `legal/COPYING.protobuf`: This file is added as `COPYING.protobuf` to the `META-INF` of protobuf projects.

Output of `show apacheSonatypeLicenseFile` 

```
[IJ]pekko > show apacheSonatypeLicenseFile
[info] pki / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] actor-testkit-typed / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] coordination / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] cluster-sharding-typed / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] actor-typed / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] docs / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] actor-typed-tests / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] testkit / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] stream / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] multi-node-testkit / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] persistence-query / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] osgi / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] persistence-typed-tests / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] persistence-testkit / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] persistence / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] cluster-tools / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] remote-tests / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] protobuf-v3 / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/legal/LICENSE.protobuf
[info] stream-tests-tck / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] protobuf / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/legal/LICENSE.protobuf
[info] actor / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] discovery / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] bench-jmh / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] cluster-metrics / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] bill-of-materials / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] stream-tests / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] stream-testkit / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] remote / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] distributed-data / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] cluster / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] actor-tests / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] slf4j / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] cluster-typed / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] serialization-jackson / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] persistence-tck / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] stream-typed / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] persistence-shared / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] persistence-typed / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] cluster-sharding / apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
[info] apacheSonatypeLicenseFile
[info] 	/Users/mdedetrich/github/incubator-pekko/LICENSE
```

mdedetrich commented on project/AddMetaInfLicenseFiles.scala:
```diff
@@ -20,19 +20,18 @@ object AddMetaInfLicenseFiles extends AutoPlugin {
   private lazy val baseDir = LocalRootProject / baseDirectory
 
   override lazy val projectSettings = Seq(
-    apacheSonatypeLicenseFile := baseDir.value / "legal" / "StandardLicense.txt",
```
This is no longer necessary because the default location of `apacheSonatypeLicenseFile` is `baseDir.value / "LICENSE"` which in this PR is now the default `LICENSE` file

pjfanning commented:
I am not all sure about this change. I woul be against changing the git root dir license, notice, disclaimer. Our git repo itself can be viewed as akin to a source release. We do need to add distinct licenses into some jars. I don't mind where they are in git as long as the jar is built ok.

pjfanning commented:
It is an ASF norm to have the source distribution LICENSE git root.

mdedetrich commented:
> I am not all sure about this change. I woul be against changing the git root dir license, notice, disclaimer. Our git repo itself can be viewed as akin to a source release. We do need to add distinct licenses into some jars. I don't mind where they are in git as long as the jar is built ok.

> It is an ASF norm to have the source distribution LICENSE git root.

These 2 sentences seem to contradict eachother? Incase its not clear, this PR does not change the resulting jar's in any way. All it is re-arranges where the license files are stored in the git repo.

Also there is a `LICENSE` file in the root, thats not being changed (its actually the same standard LICENSE that is included in the source package and in every other pekko module). The issue right now is that the LICENSE file in the root dir is an exceptional one that is only used for certain maven artifacts (i.e. the protobuff ones).

I don't know if its intentional to use the protobuff specific exclusion for the root `LICENSE` file, but thats how it is right now. If this is intentional then it should be documented because right now its extremely confusing.

pjfanning commented:
The root dir license needs to list code that is not ours. We've had other people look at this file and they seemed happy with it.

mdedetrich commented:
> The root dir license needs to list code that is not ours. We've had other people look at this file and they seemed happy with it.

Thanks for clarifying, but if thats the case then this text (for the source package) is wrong

> pekko-protobuf contains the sources of Google protobuf 2.5.0 runtime support,
moved into the source package org.apache.pekko.protobuf so as to avoid version conflicts.
For license information see COPYING.protobuf

It would either need to be updated to (emphasis is bold)

> pekko-protobuf contains the sources of Google protobuf 2.5.0 runtime support,
moved into the source package org.apache.pekko.protobuf so as to avoid version conflicts.
For license information see **licenses/COPYING.protobuf**

Or the `licenses/COPYING.protobuf` needs to be moved to the root dir.

pjfanning reviewed (changes requested):
I'd like to get more eyes on this and want to block it until then. Getting the licenses right has been one of our biggest tasks and now we are off on another tack.

pjfanning commented:
I can comment fuller later, I only have my mobile.

mdedetrich commented:
> I can comment fuller later, I only have my mobile.

No worries its not urgent, I also added the blocked label. Even if the current state is how its meant to be (including the fact the text is pointing to a wrong location), we should very clearly comment/document that those files should not be moved around.

mdedetrich commented:
Actually I just noticed that I was wrong in my previous statement about the text being wrong, it was actually correct and I was accidentally looking at the wrong branch.

Will close the PR, apologies for wasting your time.
